assignments/prototype-refactor.js

```diff
@@ -7,3 +7,106 @@ Prototype Refactor
 2. Your goal is to refactor all of this code to use ES6 Classes. The console.log() statements should still return what is expected of them.
 
 */
+class GameObject{
+
+    constructor(attributes){  
+    this.createdAt = attributes.createdAt;  
+    this.name = attributes.name;  
+    this.dimensions = attributes.dimensions  
+  }
+  
+  destroy(){  
+    return `${this.name} was removed from the game`;
+    }
+   }
+  
+   class CharacterStats extends GameObject{  
+    constructor(characterAttributes){  
+    super(characterAttributes);  
+    this.healthPoints = characterAttributes.healthPoints;  
+  } 
+     
+  takeDamage() {  
+    return `${this.name} took damage.`;  
+   }  
+  }
+  
+   
+  class Humanoid extends CharacterStats{  
+    constructor(HumanoidAttributes){  
+    super(HumanoidAttributes);  
+    this.team = HumanoidAttributes.team;  
+    this.weapons = HumanoidAttributes.weapons;  
+    this.language = HumanoidAttributes.language;  
+  }
+  
+  greet() {  
+    return `${this.name} offers a greeting in ${this.language}.`;  
+   }  
+  } 
+  
+      const mage = new Humanoid({  
+      createdAt: new Date(),  
+      dimensions: {  
+        length: 2,  
+        width: 1,  
+        height: 1,  
+      },
+  
+      healthPoints: 5,  
+      name: 'Bruce',  
+      team: 'Mage Guild',  
+      weapons: [  
+        'Staff of Shamalama',  
+      ],  
+      language: 'Common Tongue',  
+    });
+     
+    const swordsman = new Humanoid({  
+      createdAt: new Date(),  
+      dimensions: {  
+        length: 2,  
+        width: 2,  
+        height: 2,
+      },
+  
+      healthPoints: 15,  
+      name: 'Sir Mustachio',  
+      team: 'The Round Table',  
+      weapons: [  
+        'Giant Sword',  
+        'Shield',  
+      ],  
+      language: 'Common Tongue',
+  
+    });
+    
+      const archer = new Humanoid({  
+      createdAt: new Date(),  
```

Comment:
Looks like the indentation was doin a bit of an oopsie in this file. I'd highly recommend installing an extension like Prettier or Beautify to handle code formatting.